### PR TITLE
Update code to be DRY

### PR DESCRIPTION
**Pull request by ddriddle**
* This is a DRAFT
* If the team wants to go this direction then `nice.py` needs to be moved into its own repo.

### CODE REVIEW

File: src/app/nice.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    def handle_action(self, param):
+        action_id = self.get_action_identifier()
+
+        self.debug_print("action_id", self.get_action_identifier())
+
+        if action_id in self.actions.keys():
+            return self.actions[action_id](param)
+
+        return phantom.APP_SUCCESS  # TODO: Should this be an error instead?
```
Comment (edthedev) [inline src/app/nice.py:41]: Yes. I think this should be an error. "Unrecognized action was called. Did you add your action to `actions`?"

File: src/app/app.py [hunk 2]
```diff
@@ -4,130 +4,22 @@
...
 # Phantom App imports
 import phantom.app as phantom
-from phantom.base_connector import BaseConnector
 from phantom.action_result import ActionResult
 
 import requests
-import json
-from bs4 import BeautifulSoup
+
+from .nice import NiceBaseConnector, handle
```
Comment (edthedev) [inline src/app/app.py:13]: When unsure of name, consider shameless branding `IllinoisSOARBaseConnector` ?

File: src/app/nice.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# -----------------------------------------
+# Nice and dry Phantom App
+# -----------------------------------------
+# TODO: This SHOULD be split into its own repo and imported
```
Comment (edthedev) [inline src/app/nice.py:4]: Maybe we put this in a new repo next to the `deploy.py` script.

File: src/app/nice.py (hunk 1 above)
Comment (ddriddle) [reply src/app/nice.py:41]: I left it as success because that is how it was coded before, but it seemed odd.

File: src/app/nice.py (hunk 3 above)
Comment (ddriddle) [reply src/app/nice.py:4]: We need to think of a repo name, but I like the idea of packaging the helper script and the deploy.py script.

**Comment by ddriddle** [comment]
@edthedev It looks like someone may have already done something very similar:

https://pypi.org/project/phantom-dev/
https://gitlab.com/phantom6/phantom-dev

I think the author is a Splunk developer. The last release on PyPI was 9/19/2021. The last commit was around the same time. So not sure how active the project, but it might be worth taking a look at and maybe even forking and maintaining ourselves. I wonder if Daniel is aware of this project.

**Comment by edthedev** [comment]
> @edthedev It looks like someone may have already done something very similar:
> 
> https://pypi.org/project/phantom-dev/ https://gitlab.com/phantom6/phantom-dev
> 
> I think the author is a Splunk developer. The last release on PyPI was 9/19/2021. The last commit was around the same time. So not sure how active the project, but it might be worth taking a look at and maybe even forking and maintaining ourselves. I wonder if Daniel is aware of this project.

I spent a bit of time reading that code and I'm concerned that a lot has changed in Splunk SOAR in the last year.
It also feels a bit off to build on top of that much code when all we really need are two python files that we've already written `nice.py` and `deploy.py`.

**Comment by edthedev** [comment]
Need to clear this error 'E   ModuleNotFoundError: No module named 'app.deploy'' next.

**Comment by edthedev** [comment]
Progress update - new repository is created, and referenced. Need to clear some import errors, then review our changes overall.

File: src/app/nice.py (hunk 1 above)
Comment (edthedev) [reply src/app/nice.py]: This is resolved - in the other repo. - See https://github.com/techservicesillinois/phantom-toolbox/blob/5ec5a1a2d3574a140208e291c15dc0350b64ec4d/src/phtoolbox/app/base_connector.py#L40

**Review by zdc217 — APPROVED** [review]

**Review by mpitcel — APPROVED** [review]

**Comment by edthedev** [comment]
Reminder - we just resolved CLI issues in phantom-toolbox. We need to `make force-clean` and commit the updated requirements files.

**Review by tzturner — APPROVED** [review]

**Pull request merged**